### DISCUSSION TITLE
modify request in middleware and pass to path handlers

### DISCUSSION THREAD

**Discussion by ysde** [fastapi/fastapi#8204] [Questions] [answered]
**Description**

Is there a way that can modify request in middleware and pass to the path handler?

I need to get some value from header, and want to pass to path handlers so that I don't need to get the value in every path handlers.

I've tried to add the value to header. But request.header cannot be modified....

Can anyone give me some advices?

Thank you
**Additional context**

**Comment by tiangolo** [comment, marked as answer]
I assume you solved your issue, thanks for coming back to close it.
